### ISSUE TITLE
Extract embedded assemblies

### ISSUE THREAD

**Issue by Giorgi** [MiloszKrajewski/LibZ#17]
I have an app which is built with LibZ. Is there an easy way to extract embedded assemblies to disk?

**Comment by MiloszKrajewski** [comment]
"ILSpy / Resources / Save" then (usually) DeflateStream.

**Comment by Giorgi** [comment]
Yep, I guessed that too but I have many assemblies which have many other embedded assemblies and I was looking for an automated way to do it but looks like there isn't built in support for it.

**Comment by moh-hassan** [comment]
I am using inject-dll to inject a managed dll into a managed console.exe

I want to extract the embeded dll resources from the console.exe application.
I used dotpeek tool and saved the resources as .lz4 files
When I  tried to decompress using lz4 with the command
lz4 -d resource.z   resource.dll
An exception  error is fired:

>Error 44 : Unrecognized header : file cannot be decoded

What i missed to get the original embeded dll files

**Comment by MiloszKrajewski** [comment]
It is not LZ4, it is just DeflateStream.

**Comment by moh-hassan** [comment]
@MiloszKrajewski , Thank you very much.
yes, it's DeflateStream. :)
I checked the extracted files and it's typical the same as  original dll with the same identity and size.
